ex/ThirdPartyDemos/martin-doyle/04 - Interfaced Based Services/source/data.pas

```diff
@@ -0,0 +1,52 @@
+unit data;
+
+interface
+
+uses
+  {$I mormot.uses.inc}
+  mormot.core.base,
+  mormot.core.data,
+  mormot.core.json,
+  mormot.core.interfaces,
+  mormot.orm.core;
+
+const
+  HttpPort = '11111';
+
+type
+  TSample = packed record
```

Comment:
For it to work properly on FPC or oldest Delphi < 2010, we need to define the JSON serialization of this record in the initialization section below.

Reply:
Works with Delphi 7. Maybe still something wrong in my code.